solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -778,13 +910,22 @@ contract RandomBeacon is IRandomBeacon, Ownable {
             "Invalid group members"
         );
 
-        uint256 slashingAmount = relay.submitEntry(entry, group.groupPubKey);
+        uint96 slashingAmount = relay.submitEntry(entry, group.groupPubKey);
 
         if (slashingAmount > 0) {
             address[] memory groupMembersAddresses = sortitionPool
                 .getIDOperators(groupMembers);
 
-            try staking.slash(slashingAmount, groupMembersAddresses) {
+            address[] memory stakingProvidersAddresses = new address[](
+                groupMembersAddresses.length
+            );
+            for (uint256 i = 0; i < groupMembersAddresses.length; i++) {
+                stakingProvidersAddresses[i] = operatorToStakingProvider(
+                    groupMembersAddresses[i]
+                );
```

Comment:
Would be good to try doing it in one external call and compare gas usage. Same about the code in `reportRelayEntryTimeout` and `reportUnauthorizedSigning`.

Reply:
Please see https://github.com/keep-network/keep-core/pull/2903#discussion_r841539839

Reply:
Let's continue the discussion in the referenced comment. Resolving this thread as a duplicate.